src/main/java/AuthenticationService.java

```diff
@@ -6,5 +8,50 @@
  * It interfaces with the UserRepository to fetch and update user information.
  */
 public class AuthenticationService {
-    // TODO implement class
+
+    private UserRepository userRepository;
+
+    public AuthenticationService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User createUser(String username, String password) {
+        UUID id = UUID.randomUUID();
+        return new User(id, username, password);
+    }
+
+    public User changeUsername(String currentUsername, String newUsername) {
+        if (newUsername == null || newUsername.isEmpty() || newUsername.trim().isEmpty()) {
+            System.out.println("Please provide valid Username");
+            return null;
```

Comment:
Consider raising an Exception here instead of printing and returning null. Otherwise the error will get lost.